nativelib/src/main/scala/scala/scalanative/native/pure.scala

```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package native
+
+/** Let optimizer assume that given method is pure.
+  * This means that repetitive calls to the same method
+  * with the same arguments are allowed to be eliminated.
+  * Additionally, if method result to given method is not used
+  * then the call can be elided altogether.
+  */
+final class pure extends scala.annotation.StaticAnnotation
```

Comment:
These operations are primitive and known to support the language. It is therefore OK for a compiler or an optimizer to leverage its knowledge of the standard library as "intrinsics" to optimize them away. But I would refrain from exposing such a dangerous feature to developers, as long as possible.